src/pgduckdb_metadata_cache.cpp

```diff
@@ -29,17 +85,32 @@ IsExtensionRegistered() {
 		return cache.installed;
 	}
 
-	cache.installed = get_extension_oid("pg_duckdb", true) != InvalidOid;
-	cache.valid = true;
-
 	if (!callback_is_configured) {
 		callback_is_configured = true;
 		schema_hash_value = GetSysCacheHashValue1(NAMESPACENAME, CStringGetDatum("duckdb"));
 
 		CacheRegisterSyscacheCallback(NAMESPACENAME, InvalidateCaches, (Datum)0);
 	}
 
+	cache.installed = get_extension_oid("pg_duckdb", true) != InvalidOid;
+	if (cache.installed) {
+		BuildDuckdbOnlyFunctions();
+	}
+	cache.valid = true;
```

Comment:
Maybe I'm reading this wrong, but it looks like it's possible to end up with `cache.installed == false && cache.valid == true` - shouldn't the valid flag be set only in the `installed` path?

Reply:
valid == is the cache initialized. I added a bunch of comments in the file to make this much clearer.